### ISSUE TITLE
The 'tricky reference error' demo doesn't work in Safari

### ISSUE THREAD

**Issue by null-a** [probmods/webppl-editor#9]
The error isn't displayed beneath the code box as expected. The following error is logged to the console:

```
TypeError: undefined is not an object (evaluating 'e.message')
onerror - editor.js:71806
```

I'm not sure which browsers you're targeting, but I thought I should mention it.

**Comment by longouyang** [comment]
Ah, thanks. This is because Safari only passes stack traces to `catch`, not `window.onerror`

The simple fix is to just not provide this information for Safari.

A more complicated thing we could try is to associate the web trampoline runner with error handling code (either by mutation, as we do with canceling jobs, or by passing a handler as an argument)

**Comment by null-a** [comment]
I'm entirely sure this is relevant, but perhaps there are ideas in [stacktrace.js](https://github.com/stacktracejs/stacktrace.js) we can borrow here.

(HT @chrisranderson, probmods/webppl#292.)

**Comment by longouyang** [comment]
Thanks for the tip. I tried it out but stacktrace.js doesn't help us here -- it expects that an actual error object is available when `window.onerror` fires, but Safari doesn't provide this (HT http://stackoverflow.com/a/22822051/351392)

**Comment by longouyang** [comment]
In any event, I added the simple fix in dab8d4673acbf858d6899e1a1adc858f7eea0fe5, so I'll close this for now. If it's really pressing that errors look the same across browsers, we can revisit this.
